homepage/models.py

```diff
@@ -1,22 +1,39 @@
 from django.db import models
 from django.utils import timezone
 from django.conf import settings
+from django.core.validators import MinValueValidator
 
 
 class Course(models.Model):
-    course_id = models.IntegerField(primary_key=True)  # course id number as given by college
+    course_id = models.IntegerField(primary_key=True,
+                                    validators=[MinValueValidator(0)])  # course id number as given by college
     name = models.CharField(max_length=100)  # course name as given by college
     mandatory = models.BooleanField()  # is the course mandatory (True) or an elective (False)
-    credit_points = models.SmallIntegerField()  # the credit points assigned to this course by the college
+    credit_points = models.SmallIntegerField(
+        validators=[MinValueValidator(1)])  # the credit points assigned to this course by the college
     syllabi = models.CharField(max_length=200)  # link to syllabi of the course, not for use, currently
     avg_load = models.DecimalField(max_digits=6, decimal_places=5)  # average course load
     avg_rating = models.DecimalField(max_digits=6, decimal_places=5)  # average course rating
-    num_of_raters = models.IntegerField()  # number of raters for course ratnig and course load
-    num_of_reviewers = models.IntegerField()  # number of reviewers
+    num_of_raters = models.IntegerField(
+        validators=[MinValueValidator(0)])  # number of raters for course rating and course load
+    num_of_reviewers = models.IntegerField(
+        validators=[MinValueValidator(0)])  # number of reviewers
 
     def __str__(self):
         return self.name
 
+    def printDetails(self):
+        mandatory = 'yes' if self.mandatory == 1 else 'no'
+        msg = f"indentifier: {self.course_id}   \tName: {self.name}\nMandatory? {mandatory}\n"
+        msg += f"Credit Points: {self.credit_points}\nSyllabi: {self.syllabi}\n"
+        msg += f"Average Rating: {round(self.avg_rating, 3)} \tAverage Load: {round(self.avg_load, 3)}\t"
+        msg += f"{self.num_of_raters} Raters\nNumber Of Reviews: {self.num_of_reviewers}"

```

Comment:
maybe instead of repeat "msg +=..." you can do it like:
```suggestion
        msg = (
            f"Course ID: {self.course_id}   \tName: {self.name}\nMandatory? {mandatory}\n"
            f"Credit Points: {self.credit_points}\nSyllabi: {self.syllabi}\n"
            f"Average Rating: {round(self.avg_rating, 3)} \tAverage Load: {round(self.avg_load, 3)}\t"
            f"{self.num_of_raters} Raters\nNumber Of Reviews: {self.num_of_reviewers}"
        )
```